mintlify/mint
# Remove local env variables

## Summary
Added default values and refactored `process.env` such that local env variables are unnecessary for single-tenant hosting (i.e. for open-source contributors that want to make changes to client). `IS_MULTI_TENANT=true` and `ADMIN_TOKEN` are still required for multi-tenant hosting.

After this PR is merged, devs will be able to simplify their local `.env` files and there will be reduced friction to start contributing to `client/` :)

## Test Plan
1. Go into the `client/` folder.
2. Comment out your entire `.env` file.
3. Run `yarn preconfigure`, then `yarn dev`. Navigate to `http://localhost:3000` and sanity check.
4. Add back `IS_MULTI_TENANT=true` and `ADMIN_TOKEN` (but not `API_ENDPOINT`) to your `.env` file.
5. Start server locally. Navigate to `http://mintlify.localhost:3000` and sanity check. Validate `sitemap.xml` and `robots.txt`.

## TODO
- [ ] Reorganize/remove `lib/` folder and remove exception from `.gitignore`
- [ ] Clean up Infisical secrets in Mintlify Client
- [ ] Clean up Vercel env variables
- [ ] Add instructions (internal docs) for multi-tenant local hosting
